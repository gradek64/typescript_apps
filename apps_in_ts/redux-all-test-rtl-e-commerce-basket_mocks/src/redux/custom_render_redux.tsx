// test-utils.js
import React, { ReactElement } from 'react'
import { Provider, } from 'react-redux';
import { createStore, combineReducers } from 'redux'
import { render } from '@testing-library/react'
import cartReducer from '../redux/reducers/cart_reducer'
import { basketItemsInterface } from '../redux/reducers/cart_reducer'


export interface ICart {
  cart: {
    items: basketItemsInterface[],
    numberOfItems: number,
  }
}
export function renderWithRedux(
  component: ReactElement,
  options: {
    state: ICart,
    [key: string]: any,
  }
) {
  const rootReducer = combineReducers({
    cart: cartReducer,
  })
  const store = createStore(rootReducer, options.state);
  const AllProviders: React.FC<{}> = ({ children }) =>
    <Provider store={store}>
      {children}
    </Provider>


  const wrappedComponent = render(component, {
    wrapper: AllProviders,
    ...options
  });

  return { store, ...wrappedComponent };
}
