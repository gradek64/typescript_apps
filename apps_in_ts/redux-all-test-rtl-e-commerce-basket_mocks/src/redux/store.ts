import { createStore, combineReducers } from 'redux';
import cartReducer from './reducers/cart_reducer';

const rootReducer = combineReducers({
  cart: cartReducer,
})
const store = createStore(
  rootReducer,
);

export default store;
export type RootState = ReturnType<typeof rootReducer>

