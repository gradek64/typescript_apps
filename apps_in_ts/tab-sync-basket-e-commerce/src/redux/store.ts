import { createStore, combineReducers,applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import cartReducer from './reducers/cart_reducer';
import shipping_destination_reducer from './reducers/shipping_destination_reducer';

const rootReducer = combineReducers({
  cart: cartReducer,
  shippingCountry:shipping_destination_reducer
})
const store = createStore(
  rootReducer,
  applyMiddleware(thunk),
);

export default store;
export type RootState = ReturnType<typeof rootReducer>

