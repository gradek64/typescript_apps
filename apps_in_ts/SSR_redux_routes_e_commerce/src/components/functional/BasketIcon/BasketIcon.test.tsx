import React from 'react';
import { renderWithRedux } from '../../../redux/custom_render_redux'
import { fireEvent } from '@testing-library/react'

import WrapperBasketIcon from './BasketIcon';
import { ICart } from '../../../redux/custom_render_redux'

const props = {
  eventHandler: jest.fn(),
};

const cart: ICart = {
  cart: {
    items: [],
    numberOfItems: 0,
  }
};

const cartWithOneItem: ICart = {
  cart: {
    items: [{
      title: 'Russell Hobbs Pennine Illuminating S Steel Kettle 20444',
      price: 24.99,
      productId: 5493179,
    }],
    numberOfItems: 1,
  }
};

const getDOMElements = (container: Element) => {
  const addToBasketButton = container.querySelector('button')
  return { addToBasketButton }
}

describe('render', () => {
  it('renders with no items number', () => {
    const { asFragment } = renderWithRedux(<WrapperBasketIcon {...props} />, { state: cart });
    expect(asFragment()).toMatchSnapshot();
  });
  it('renders with 1 items number', () => {
    const { asFragment } = renderWithRedux(<WrapperBasketIcon {...props} />, { state: cartWithOneItem });
    expect(asFragment()).toMatchSnapshot();
  });
})
describe('events', () => {
  it('BasketIcon does not trigger when basket is empty', () => {
    const { eventHandler } = props
    const { container } = renderWithRedux(<WrapperBasketIcon {...props} />, { state: cart });
    const { addToBasketButton } = getDOMElements(container)
    fireEvent.click(addToBasketButton as Element)
    expect(eventHandler).not.toHaveBeenCalled()
  })
  it('BasketIcon trigger when basket has at least 1 item', () => {
    const { eventHandler } = props
    const { container } = renderWithRedux(<WrapperBasketIcon {...props} />, { state: cartWithOneItem });
    const { addToBasketButton } = getDOMElements(container)
    fireEvent.click(addToBasketButton as Element)
    expect(eventHandler).toHaveBeenCalled()
  })
})
