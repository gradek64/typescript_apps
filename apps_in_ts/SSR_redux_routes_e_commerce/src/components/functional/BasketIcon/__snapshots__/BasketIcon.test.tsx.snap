// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders with 1 items number 1`] = `
<DocumentFragment>
  <div
    class="basket-icon sticky"
  >
    <button
      class="pointer"
      role="link"
    >
      <img
        alt="open basket"
        src="/images/shopping-cart-3.svg"
      />
    </button>
    <span
      class="inBasket"
    >
      1
    </span>
  </div>
</DocumentFragment>
`;

exports[`render renders with no items number 1`] = `
<DocumentFragment>
  <div
    class="basket-icon sticky"
  >
    <button
      class="no-pointer"
      disabled=""
      role="link"
    >
      <img
        alt="open basket"
        src="/images/shopping-cart-3.svg"
      />
    </button>
  </div>
</DocumentFragment>
`;
